src/index.ts

```diff
@@ -132,6 +132,9 @@ export interface IlpPluginBtpConstructorOptions {
     port: number,
     secret: string
   },
+  raw?: {
```

Comment:
> Is there a reason for this extra layer of indirection?

Good suggestion. I think `listener.socket` makes the most sense.

> Also, from the option alone it isn't clear whether the created plugin will act as a client or server.

The plugin is not going to act as a WS server because it is created around an existing ws socket so it's not listening for new connections.

BUT it will handle incoming auth messages (i.e. it's a "BTP server" in as much as the difference between a client and server is that one performs auth with the other even though they may actually be peers)

This is a general problem (I think) with the architecture of the plugin in that it mixes the transport (websockets) with the protocol (BTP).

I tried a more decoupled approach with `ilp-transport`, interested to hear what you think of that and if we could move to something similar with BTP: https://github.com/adrianhopebailie/ilp-transport